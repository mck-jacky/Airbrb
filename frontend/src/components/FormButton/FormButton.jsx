import React from 'react'
import PropTypes from 'prop-types';
import Button from '@mui/material/Button';

const FormButton = ({ text, name = '' }) => {
  return (
    <Button data-testid='form-button'
      type="submit"
      name={name}
      fullWidth
      variant="contained"
      color="primary"
      aria-label={text}
      sx={{ mt: 3, mb: 2 }}
    >
      {text}
    </Button>
  )
}

FormButton.propTypes = {
  text: PropTypes.string,
  name: PropTypes.string
}

export default FormButton
