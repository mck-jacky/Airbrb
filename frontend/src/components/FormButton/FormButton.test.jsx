import React from 'react';
import { render, screen } from '@testing-library/react';
import FormButton from './FormButton';
import ShallowRenderer from 'react-test-renderer/shallow'

describe('FormButton', () => {
  it('is in the document', () => {
    render(<FormButton text="123" />)
    const component = screen.getByTestId('form-button')
    expect(component).toBeInTheDocument();
  })

  it('has same text', function () {
    render(<FormButton text="123" />)
    const component = screen.getByTestId('form-button')
    expect(component).toHaveTextContent('123')
  });

  it('has to match the snapshot', () => {
    const renderer = ShallowRenderer.createRenderer()
    renderer.render(<FormButton text="123" />)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
})
