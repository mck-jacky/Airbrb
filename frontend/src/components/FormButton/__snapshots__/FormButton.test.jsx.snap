// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormButton has to match the snapshot 1`] = `
<ForwardRef(Button)
  aria-label="123"
  color="primary"
  data-testid="form-button"
  fullWidth={true}
  name=""
  sx={
    Object {
      "mb": 2,
      "mt": 3,
    }
  }
  type="submit"
  variant="contained"
>
  123
</ForwardRef(Button)>
`;
